source/extensions/filters/network/thrift_proxy/binary_protocol_impl.cc

```diff
@@ -265,6 +265,106 @@ bool BinaryProtocolImpl::readBinary(Buffer::Instance& buffer, std::string& value
   return readString(buffer, value);
 }
 
+void BinaryProtocolImpl::writeMessageBegin(Buffer::Instance& buffer, const std::string& name,
+                                           MessageType msg_type, int32_t seq_id) {
+  BufferHelper::writeU16(buffer, Magic);
+  BufferHelper::writeU16(buffer, static_cast<uint16_t>(msg_type));
+  writeString(buffer, name);
+  BufferHelper::writeI32(buffer, seq_id);
+}
+
+void BinaryProtocolImpl::writeMessageEnd(Buffer::Instance& buffer) {
+  UNREFERENCED_PARAMETER(buffer);
+}
+
+void BinaryProtocolImpl::writeStructBegin(Buffer::Instance& buffer, const std::string& name) {
+  UNREFERENCED_PARAMETER(buffer);
+  UNREFERENCED_PARAMETER(name);
+}
+
+void BinaryProtocolImpl::writeStructEnd(Buffer::Instance& buffer) {
+  UNREFERENCED_PARAMETER(buffer);
+}
+
+void BinaryProtocolImpl::writeFieldBegin(Buffer::Instance& buffer, const std::string& name,
+                                         FieldType field_type, int16_t field_id) {
+  UNREFERENCED_PARAMETER(name);
+
+  BufferHelper::writeI8(buffer, static_cast<uint8_t>(field_type));
+  if (field_type == FieldType::Stop) {
+    return;
+  }
+
+  BufferHelper::writeI16(buffer, field_id);
+}
+
+void BinaryProtocolImpl::writeFieldEnd(Buffer::Instance& buffer) { UNREFERENCED_PARAMETER(buffer); }
+
+void BinaryProtocolImpl::writeMapBegin(Buffer::Instance& buffer, FieldType key_type,
+                                       FieldType value_type, uint32_t size) {
+  if (size > INT32_MAX) {
+    throw EnvoyException(fmt::format("illegal binary protocol map size {}", size));
+  }
+
+  BufferHelper::writeI8(buffer, static_cast<int8_t>(key_type));
+  BufferHelper::writeI8(buffer, static_cast<int8_t>(value_type));
+  BufferHelper::writeI32(buffer, static_cast<int32_t>(size));
```

Comment:
Should we verify that `size` is nonnegative? (Same question applies to `writeListBegin` and `writeSetBegin`)

Reply:
There's a comparison of size against INT32_MAX (which is 0x7fffffff) which guarantees that `static_cast<int32_t>(size)` is non-negative.